### ISSUE TITLE
Bug with array of @oneOf

### ISSUE THREAD

**Issue by yujonglee** [graphql-hive/envelop#1335]
### Issue workflow progress

_Progress of the issue based on the [Contributor Workflow](https://github.com/the-guild-org/Stack/blob/master/CONTRIBUTING.md#a-typical-contributor-workflow)_

- [x] 1. The issue provides a [minimal reproduction](https://en.wikipedia.org/wiki/Minimal_reproducible_example) available on [Stackblitz](https://stackblitz.com/fork/node). -> in Github
  - _Please install the latest `@envelop/*` packages that you are using._
  - _Please makee sure the reproduction is as small as possible._
- [ ] 2. A failing test has been provided
- [ ] 3. A local solution has been provided
- [ ] 4. A pull request is pending review

---

**Describe the bug**
`@oneOf` validation doesn't work properly with array.
1. When type is `input: [OneOf]!`, `[]` throws error. I expect this to happen for `input: [OneOf!]!`, but not `input: [OneOf]!`
2. Validation is not applied to each element of array. 

**To Reproduce**

Clone my repo.  Sorry, but I don't know how to use `Stackblitz` properly.
https://github.com/yujong-lee/oneof-reproduce

`npm run start (localhost:3000)`

1.
```graphql 
query {
  test(input: [])
}
``` 

2.
```graphql 
query {
  test(input: [{str1: "s", str2: "s"}])
}
``` 
**Expected behavior**

1. For 1,
```
{
  "errors": [
    {
      "message": "Exactly one key must be specified for input type \"OneOf\"",
```
This should not happen

2. For 2, `oneOf` error should be thrown.

**Environment:**

- OS: Mac 12.1
- NodeJS: v16.13.2
  - "@envelop/core": "^2.1.0",
  - "@envelop/extended-validation": "^1.4.0",

**Additional context**

**Comment by dotansimha** [comment]
@n1ru4l can you please take a look? :)

**Comment by n1ru4l** [comment]
@yujong-lee this is addressed by https://github.com/dotansimha/envelop/pull/1336

**Comment by dotansimha** [comment]
Fixed in `@envelop/extended-validation@1.4.1`. Thanks @n1ru4l

**Comment by yujonglee** [comment]
Thank you so much.
@dotansimha  @n1ru4l
